motr/ut/obj.c

```diff
@@ -658,6 +658,35 @@ static void ut_test_entity_namei_op_delete(void)
 	ut_entity_namei_op(M0_EO_DELETE);
 }
 
+static void ut_test_obj_lid_assign(void)
+{
+	struct m0_obj     obj;
+	struct m0_entity  ent;
+	struct m0_realm   realm;
+	struct m0_op     *ops[1] = {NULL};
+	struct m0_client *instance = NULL;
+	int               rc;
+
+	/* init */
+	rc = ut_m0_client_init(&instance);
+	M0_UT_ASSERT(rc == 0);
+
+	ut_realm_entity_setup(&realm, &ent, instance);
+	ent.en_realm = &realm;
+
+	memset(&obj, 0, sizeof obj);
+	obj.ob_attr.oa_buf_size = 40960;
+	m0_obj_init(&obj, &realm, &ent.en_id, 0);
+
+	rc = m0_entity_create(NULL, &obj.ob_entity, &ops[0]);
+	M0_UT_ASSERT(obj.ob_attr.oa_layout_id > 0 &&
+			obj.ob_attr.oa_layout_id < 15);
+	M0_UT_ASSERT(rc = -ENOENT);
```

Comment:
This is a bug.
```
M0_UT_ASSERT(rc == -ENOENT);
```
?

Reply:
Ah, seems like I copied this bug along with the UT below this, I'll fix this and other UTs affected by this.

Reply:
will have to change the check to 
```
M0_UT_ASSERT(rc == -EINVAL)
```
as `m0__obj_layout_instance_build()` returns -EINVAL.
```
        layout = m0_layout_find(&cinst->m0c_reqh.rh_ldom, layout_id);
        if (layout == NULL) {
                rc = -EINVAL;
                goto out;
        }
```

Reply:
OK